storage/migration.md

```diff
@@ -0,0 +1,192 @@
+
+__MariaDB example output__:
+```shell
+$ kubectl get pv pvc-647af707-d53b-4dd3-9571-426a4d321593
+NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                 STORAGECLASS                    REASON   AGE
+pvc-647af707-d53b-4dd3-9571-426a4d321593   1Gi        RWO            Retain           Bound    default/mariadb-pvc   upcloud-block-storage-maxiops            32s
+```
+Reclaim policy column tells that we have `retain` policy and we can continue. Persistent volume claim name (`mariadb-pvc`) and namespace (`default`) are shown in the claim column.
+
+## Export PV and PVC objects
+Export persistent volume and persistent volume claim objects from the old cluster. Note that PVC objects are namespaced and PV objects are not.
+```shell
+$ kubectl get pv <pv_name> -o yaml > <pv_name>.yaml
+$ kubectl -n <namespace> get pvc <pvc_name> -o yaml > <pvc_name>.yaml
+```
+*check that <pv_name>.yaml and <pvc_name>.yaml files contains PV and PVC objects before continuing*   
+
+__MariaDB example output__:
+```shell
+$ kubectl get pv pvc-647af707-d53b-4dd3-9571-426a4d321593 -o yaml > pvc-647af707-d53b-4dd3-9571-426a4d321593.yaml
+$ kubectl -n default get pvc mariadb-pvc -o yaml > mariadb-pvc.yaml
+```
+
+## Shutdown application(s)
+Shutdown application that uses PVC in the old cluster. This will detach storage device from worker node. 
+
+__MariaDB example output__:
+```shell
+$ kubectl -n default scale deployment mariadb --replicas=0
+deployment.apps/mariadb scaled
+```
+
+Before continuing, login to Hub and check that the storage device with the same name as persistent volume, isn't attached to any server. 
+
+## Determine storage device ID
+Determine underlying storage device's `uuid` so that correct permissions can be granted for the new cluster. Take a copy of the storage `uuid` value.
+Storage `uuid` can be checked from Hub (using PV name) or using command:
+```shell
+$ kubectl get pv <pv_name> -o custom-columns="storage_uuid":.spec.csi.volumeHandle,"pv_name":.metadata.name
+```
+
+__MariaDB example output__:
+```shell
+$ kubectl get pv pvc-647af707-d53b-4dd3-9571-426a4d321593 -o custom-columns="storage_uuid":.spec.csi.volumeHandle,"pv_name":.metadata.name
+storage_uuid                           pv_name
+01a1990f-3409-4cad-ba87-f7925f149842   pvc-647af707-d53b-4dd3-9571-426a4d321593
+```
+
+---
+
+# New cluster
+Set `KUBECONFIG` environment variable to point to the new cluster config.  
+
+## Grant device permissions
+CSI driver is run using sub-account credentials. Get CSI user's username in the new cluster:
+```shell
+$ kubectl -n kube-system get secrets upcloud -o yaml -o jsonpath='{.data.username}'|base64 -d
+```
+
+### Grant device permissions using Hub
+Go to https://hub.upcloud.com/people/permissions and grant CSI sub-account for permission to access storage.
+
+### Grant device permissions using API
+Use following permission JSON object to set  permission:  
```

Comment:
```suggestion
Use following permission JSON object to set permission:  
```

Reply:
@thevilledev thanks for review :+1: I'll push fixes.